src/shared/sleep-config.c

```diff
@@ -229,6 +245,51 @@ int find_hibernate_location(char **device, char **type, uint64_t *size, uint64_t
                                 continue;
                         }
 
+                        /* Check if swap matches the resume= and resume_offset= values if configured. */
+                        if (!streq(resume_offset, "0") && !streq(resume, "0:0")) {
+                                _cleanup_close_ int fd;
```

Comment:
Everything declared with `_cleanup_` must be initialized (except for some cases in test code, but that is not relevant here). I.e. you need `= -1;` here.